templates/tftmpl/tmplfunc/nodes.go

```diff
@@ -0,0 +1,140 @@
+package tmplfunc
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	consulapi "github.com/hashicorp/consul/api"
+	"github.com/hashicorp/go-bexpr"
+	"github.com/hashicorp/hcat"
+	"github.com/hashicorp/hcat/dep"
+	"github.com/pkg/errors"
+)
+
+func nodesFunc(recall hcat.Recaller) interface{} {
```

Comment:
Although these functions are not public, can you add code comments for them since they contain a lot of broader context that is not immediately noticeable.

You can follow the general format found here https://github.com/hashicorp/consul-terraform-sync/blob/afe4d8d1225378fa81fcb5f7dad37ab71fa5295d/templates/tftmpl/tmplfunc/catalog_services_registration.go#L15-L22